table/index_builder.cc

```diff
@@ -146,14 +146,38 @@ Status PartitionedIndexBuilder::Finish(
   }
 }
 
+// Estimate size excluding the top-level index
+// It is assumed that this method is called before writing index partition
+// starts
 size_t PartitionedIndexBuilder::EstimatedSize() const {
   size_t total = 0;
   for (auto it = entries_.begin(); it != entries_.end(); ++it) {
     total += it->value->EstimatedSize();
   }
-  total += index_block_builder_.CurrentSizeEstimate();
```

Comment:
It returns 8 because at this point the top-level index is not full yet. I removed the line and clarified what the function does in comments.